src/plugins/legacy/polygonRoi/polygonRoiToolBox.cpp

```diff
@@ -59,16 +68,54 @@ polygonRoiToolBox::polygonRoiToolBox(QWidget *parent ) :
     repulsorTool->setCheckable(true);
     connect(repulsorTool,SIGNAL(clicked(bool)),this,SLOT(activateRepulsor(bool)));
 
+    specialities = new QComboBox();
+
+    structureList = new QListWidget();
+    structureList->setDragDropMode(QAbstractItemView::InternalMove);
+    structureList->setDragEnabled(true);
+    structureList->setAcceptDrops(true);
+    structureList->setDropIndicatorShown(true);
+    structureList->hide();
+    plusButton = new QPushButton(QIcon(":/pixmaps/plus-button.png"), "");
+    plusButton->setMaximumSize(QSize(20,20));
+    connect(plusButton, SIGNAL(pressed()), this, SLOT(addLabelName()));
+    plusButton->hide();
+
+    applyButton = new QPushButton("Apply");
+    applyButton->setProperty("default", QVariant(true));
+    connect(applyButton, SIGNAL(pressed()), this, SLOT(setPredefinedLabelNames()));
+
+    QStringList spe;
+    spe << "default" << "prostate";
+    specialities->addItems(spe);
+    connect(specialities, SIGNAL(currentIndexChanged(int)), this, SLOT(updateListOfLabelNames(int)));
+
     currentView = nullptr;
 
     QHBoxLayout *ButtonLayout0 = new QHBoxLayout();
     layout->addLayout( ButtonLayout0 );
     ButtonLayout0->addWidget(addNewCurve);
 
-    QHBoxLayout *ButtonLayout1 = new QHBoxLayout();
-    layout->addLayout( ButtonLayout1 );
-    ButtonLayout1->addWidget(interpolate);
-    ButtonLayout1->addWidget(repulsorTool);
+    QVBoxLayout *ButtonLayout1 = new QVBoxLayout();
```

Comment:
i'm not a big fan of "button1", "button2", etc names for widget. And as you saw, if we add a new group, we need to change every following number. I would prefer something like "layoutSpecialities" or "structureListLayout", etc ^^